### DISCUSSION TITLE
GZip log files corrupted upon download

### DISCUSSION THREAD

**Discussion by GrizzlyAK** [zwave-js/zwave-js-ui#1711] [Q&A]
I discovered this by accident, and kpine confirmed it, but if you click on a filename in the Store Explorer and click on the download option that appears, the file that is downloaded is either empty or corrupt. However, if you follow the method outlined in troubleshooting [here](https://zwave-js.github.io/zwavejs2mqtt/#/troubleshooting/bug_report?id=download-zip) (which kpine pointed out to me), by selecting the file's check box, then clicking the cog, the download is fine. Just an FYI.

**Comment by robertsLando** [comment]
The two downloads methods are different as one (zip download) is done server side and the other is done client side by just doenloading the selected file as text. Could you tell me the extension of the file you are tring to download? Does this happen with all kind of files?

**Reply by GrizzlyAK** [reply]
Ah, that probably explains it. I was trying to download the GZip backup files for the zwave-js logs. Clicking on the files seems to download _something_, but it can't be opened with any zip program. Probably because it is a text file, as you say. Not a big deal, it was just confusing me. In the past I used the check method, but for some reason this time, I clicked on the file and saw the download link.  Thanks.

**Reply by kpine** [reply]
It's the "previewer" downloader that's the problem. When you click on a gzip file, it gives an error that gzip files aren't supported, but it shows "Save" and "Download" options anyways. I assume that since the preview window is empty, the download option is just downloading the empty preview (the download file is 0 bytes), instead of the original file.

Perhaps the preview controls should not be shown for unsupported file types?

![image](https://user-images.githubusercontent.com/791794/134261566-ddd55d1c-7ee0-4d4a-b02b-31c00238a2fe.png)

**Reply by robertsLando** [reply]
> Perhaps the preview controls should not be shown for unsupported file types?

Yeah that's the problem, fix is coming

**Reply by robertsLando** [reply]
6b7d63ebdf4013e10d9135d7610422616a71b5e8
